velox/functions/prestosql/DateTimeFunctions.h

```diff
@@ -865,6 +872,22 @@ struct DateTruncFunction : public TimestampWithTimezoneSupport<T> {
       return;
     }
 
+    if (timeZone_ == nullptr) {
```

Comment:
is this optimization ok if timeZone_ is not null?

Reply:
Time zones can complicate this issue. For UTC standard time zones like UTC-05:00, we only need to add a fixed offset in the calculation. However, for time zone representations like America/New_York, which consider daylight saving, a more complex conversion is required.

Reply:
How about using getSeconds(timestamp, timeZone) ?

Reply:
@lxian I don't know there is a getSeconds in velox. And even with this function, it can only replace date_trunc('second', timestamp) accrording to its name, right?

Reply:
@lxian using `getSeconds(timestamp, timeZone) ` to support timezone as you said, could you take a look again.